docs/src/main/tut/add-support-for-complex-types.md

```diff
@@ -0,0 +1,246 @@
+support, the [second part](#add-support-for-identifiable) provides the three methods to add support for that complex type and the
+[final part](#summary) discusses when one method should be picked instead of another one.
+
+#### Data type to support
+
+First of all, let's start with the data type that we want to support. 
+The example is composed by one interface and two implementations:
+
+```tut:silent
+trait Identifiable {
+  def getId: String
+}
+
+class Class1(id: String) extends Identifiable {
+  def getId: String = id
+}
+
+class Class2(id: String, value: Int) extends Identifiable {
+  def getId: String = id
+  def getValue: Int = value
+}
+```
+
+Because PureConfig is based on Shapeless and Shapeless doesn't support non-case classes and interfaces,
+the class above cannot be read automatically by PureConfig.
+
+#### Add support for Identifiable
+
+##### Method 1: Implement `ConfigReader[Identifiable]` manually
+
+Similarly to add support for simple types, it is possible to manually create a
+`ConfigReader[Identifiable]`:
+
+```tut:silent
+import com.typesafe.config._
+import pureconfig._
+import pureconfig.error._
+
+def extractId(obj: ConfigObject): Either[ConfigReaderFailures, String] =
+  Option(obj.get("id")) match {
+    case None =>
+      Left(ConfigReaderFailures(KeyNotFound("id", ConfigValueLocation(obj))))
+    case Some(id) =>
+      ConfigReader[String].from(id)
+  }
+
+def extractValue(obj: ConfigObject): Either[ConfigReaderFailures, Int] =
+  Option(obj.get("value")) match {
+    case None =>
+      Left(ConfigReaderFailures(KeyNotFound("value", ConfigValueLocation(obj))))
+    case Some(value) =>
+      ConfigReader[Int].from(value)
+  }
+
+implicit val identifiableConfigReader = ConfigReader.fromFunction({
+case obj: ConfigObject =>
+ Option(obj.get("type")) match {
+   case None =>
+     Left(ConfigReaderFailures(KeyNotFound("type", ConfigValueLocation(obj))))
+   case Some(t) if t.valueType() == ConfigValueType.STRING =>
+     t.unwrapped().asInstanceOf[String] match {
+       case "class1" => extractId(obj).right.map(new Class1(_))
+       case "class2" =>
+         for {
+           id <- extractId(obj).right
+           value <- extractValue(obj).right
+         } yield new Class2(id, value)
+       case x =>
+         Left(ConfigReaderFailures(CannotConvert(obj.toString,
+           "Identifiable", s"type has value $t instead of class1 or class2",
+           ConfigValueLocation(obj),
+           "")))
+     }
+   case Some(t) =>
+     Left(ConfigReaderFailures(CannotConvert(obj.toString,
+       "Identifiable", s"type has value $t instead of class1 or class2",
+       ConfigValueLocation(obj),
+       "")))
+ }
+case wrong =>
+  Left(ConfigReaderFailures(WrongType(wrong.valueType(), Set(ConfigValueType.OBJECT), ConfigValueLocation(wrong), "")))
+})
+```
+
+**pros**: The main feature of this method is that it take control over PureConfig on how to
```

Comment:
This reads a bit awkwardly to me. It's trying to express something English isn't terribly good at, IMHO. 
I'd suggest 
- "... that PureConfig delegates to it when decoding your configuration"
or
- "... that it supersedes PureConfig to decode your configuration"

If you prefer your wording, then s/take/takes/